src/main/java/com/google/api/codegen/util/ProtoParser.java

```diff
@@ -86,6 +88,57 @@ public ProtoParser(boolean enableProtoAnnotations) {
     }
   }
 
+  @Nullable
+  /* Return the name of the field representing the header parameter. */
+  public List<String> getHeaderParams(Method method) {
```

Comment:
Please check the commend on L117. But in general, do we really need a list here? 

Also, can this method return just an empty list instead of `null` if nothing is found? If we do not need to distinguish between emtpy list and `null` it makes usage of this method easier (since caller does not have to handle `null`s, like it is done [here](https://github.com/googleapis/gapic-generator/pull/2491/files#diff-a81b001e7c99214072d01c43e97184bbR285).

Reply:
Ok, returning an empty collection (changed to explicit ImmutableSet)